Source/Particles/ElementaryProcess/QEDInternals/BreitWheelerEngineWrapper.H

```diff
@@ -85,71 +96,73 @@ class BreitWheelerEvolveOpticalDepth
+     * If the  quantum parameter parameter of the photon is
+     * < bw_minimum_chi_phot, the method returns immediately.
+     * The method returns also if the energy of the photon is insufficient
+     * to generate a pair.
+     * @param[in] ux,uy,uz gamma*v components of the photon.
      * @param[in] ex,ey,ez electric field components (SI units)
      * @param[in] bx,by,bz magnetic field components (SI units)
      * @param[in] dt timestep (SI units)
-     * @param[in,out] opt_depth optical depth of the photon. It is modified by the method.
-     * @return a flag which is 1 if optical depth becomes negative (i.e. a pair has to be generated).
+     * @param[in,out] opt_depth optical depth of the photon.
+     * @return a flag which is 1 if chi_phot was out of table
      */
-    AMREX_GPU_HOST_DEVICE
+    AMREX_GPU_DEVICE
     AMREX_FORCE_INLINE
     int operator()(
-    amrex::Real px, amrex::Real py, amrex::Real pz,
-    amrex::Real ex, amrex::Real ey, amrex::Real ez,
-    amrex::Real bx, amrex::Real by, amrex::Real bz,
-    amrex::Real dt, amrex::Real& opt_depth) const noexcept
+        const amrex::Real ux, const amrex::Real uy, const amrex::Real uz,
+        const amrex::Real ex, const amrex::Real ey, const amrex::Real ez,
+        const amrex::Real bx, const amrex::Real by, const amrex::Real bz,
+        const amrex::Real dt, amrex::Real& opt_depth) const noexcept
     {
-        bool has_event_happened{false};
-
-        //the library provides the time (< dt) at which the event occurs, but this
-        //feature won't be used in WarpX for now.
-        amrex::Real unused_event_time{0.0};
-
-        PicsarBreitWheelerEngine::
-        internal_evolve_opt_depth_and_determine_event(
-            px, py, pz,
-            ex, ey, ez,
-            bx, by, bz,
-            dt, opt_depth,
-            has_event_happened, unused_event_time,
-            m_dummy_lambda,
-            picsar::multi_physics::lookup_1d<amrex::Real>{
-                m_TTfunc_size,
-                m_p_TTfunc_coords,
-                m_p_TTfunc_data},
-            m_ctrl);
-
-        return has_event_happened;
+        namespace pxr_m = picsar::multi_physics::math;
+        namespace pxr_p = picsar::multi_physics::phys;
+        namespace pxr_bw = picsar::multi_physics::phys::breit_wheeler;
+
+        constexpr amrex::Real m_e = PhysConst::m_e;
+        const auto energy = std::sqrt(ux*ux + uy*uy + uz*uz)*m_e*PhysConst::c;
+
+        const auto chi_phot = QedUtils::chi_photon(
+            m_e*ux, m_e*uy, m_e*uz, ex, ey, ez, bx, by, bz);
+
+        //Optical depth is not evolved for photons having less energy than what is
+        //required to generate a pair or a quantum parameter smaller than
+        //m_bw_minimum_chi_phot
+        const auto gamma_photon = std::sqrt(ux*ux + uy*uy + uz*uz);
+        if (gamma_photon < pxr_m::two<amrex::Real> ||
+            chi_phot < m_bw_minimum_chi_phot)
+            return 0;
```

Comment:
```suggestion
        const auto u_norm = std::sqrt(ux*ux + uy*uy + uz*uz);
        const auto energy = u_norm*m_e*PhysConst::c;

        const auto chi_phot = QedUtils::chi_photon(
            m_e*ux, m_e*uy, m_e*uz, ex, ey, ez, bx, by, bz);

        //Optical depth is not evolved for photons having less energy than what is
        //required to generate a pair or a quantum parameter smaller than
        //m_bw_minimum_chi_phot
        const auto gamma_photon = u_norm/PhysConst::c;
        if (gamma_photon < pxr_m::two<amrex::Real> ||
            chi_phot < m_bw_minimum_chi_phot)
            return 0;
```

Slight refactoring to have less calculation and also fix error with unit of gamma.

Reply:
Thanks a lot for having spotted this bug 🐛 !

Reply:
Thanks a lot for having spotted this bug 🐛  !